src/nfts/nftCard.tsx

```diff
@@ -12,26 +13,64 @@ export function NftCover({
   className?: string;
   nft: Nft | undefined;
 }) {
-  const [isLoading, setLoading] = useState(true);
-  const isVideo = nft?.foreground ? /\.mp4$/.test(nft?.foreground) : false;
+  const [data, setData] = useState<{
+    url: string;
+    contentType: string | null;
+  } | null>(null);
+  const [isLoading, setLoading] = useState(false);
 
-  return isVideo ? (
-    <video
-      autoPlay
-      loop
-      className={clsx(styles.cover, className, isLoading && 'skeleton-glow')}
-      // eslint-disable-next-line react/no-unknown-property
-      onLoad={() => nft?.foreground && setLoading(false)}
-    >
-      <source src={nft?.foreground} type="video/mp4" />
-    </video>
-  ) : (
-    <img
-      src={nft?.foreground}
-      className={clsx(styles.cover, className, isLoading && 'skeleton-glow')}
-      style={nft?.background}
-      onLoad={() => nft?.foreground && setLoading(false)}
-    />
+  useEffect(() => {
+    if (nft?.foreground) {
+      setLoading(true);
+
+      fetch(nft.foreground)
+        .then(response => (response.ok ? response : Promise.reject()))
```

Comment:
I'd say it's better to avoid mixing `async/await` with `.then/.catch`. But to use `async/await` I would do `Promise.resolve().then(async ...` and do everything in that function, without following `.catch/.finally`, instead using `try/catch/finally` in an async function. In general either one way or another, without mixing, except initial `Promise.resolve`, just because you can't use `async/await` inside `useEffect`.

Also, empty reject doesn't make any sense, so it makes sense to use `throw` in any case, with `async/await` or not. Because you can't throw "nothing", but you can reject with nothing.